shared-module/audiomixer/Mixer.c

```diff
@@ -118,18 +135,45 @@ static inline uint32_t mult16signed(uint32_t val, int32_t mul) {
     asm volatile ("ssat %0, %1, %2, asr %3" : "=r" (hi) : "I" (bits), "r" (hi), "I" (shift));
     asm volatile ("pkhbt %0, %1, %2, lsl #16" : "=r" (val) : "r" (lo), "r" (hi)); // pack
     return val;
+    #else
+    uint32_t result = 0;
+    float mod_mul = (float)mul / (float)((1 << 15) - 1);
```

Comment:
Is it possible to do this without a float? Float math will be slower than integer math.

Reply:
I agree.

Reply:
Maybe. I fixed the errors so it now works and seems acceptably fast on the RP2040. I think with a bit of work the above ASM may be able to be used without the `float` being used, but in that case you may need an int64 which I'm not sure limits us on some boards or not? I did upload the fixed code so it can at least work, not sure if its worth continuing on the potential speed improvement or PRing this now?

Reply:
This is fine for now. Someone else can update it later if they want.